File: Tests/c/two.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    int processor_id;
    int *multipliers;
    int count;
} DataPipeline;

int* get_default_multipliers() {
    int default_arr[] = {2, 3, 4};
    return default_arr;
}

void init_pipeline(DataPipeline *pipeline) {
    pipeline->processor_id = 101;
    pipeline->count = 3;
    pipeline->multipliers = get_default_multipliers();
}

int main() {
    DataPipeline pipe;
    init_pipeline(&pipe);

    printf("Inicializējam datus...\n");

    if (pipe.multipliers[0] != 2) {
        printf("Kļūda: Pirmā reizinātāja vērtība ir %d, nevis 2!\n", pipe.multipliers[0]);
        return 1;
    }

    printf("Scenārijs 2 darbojas veiksmīgi.\n");
    return 0;
}

File: Tests/python/two.py
class DataPipeline:

	def __init__(self):
		self._processors = []

	def add_multiplier_stages(self, multipliers: list[int]):
		for m in multipliers:
			processor_name = f"Multiply_by_{m}"
			self._processors.append(lambda x: x * m)

	def process(self, data: list[int]) -> list[list[int]]:
		results = []
		for func in self._processors:
			results.append([func(item) for item in data])
		return results


if __name__ == "__main__":
	pipeline = DataPipeline()
	pipeline.add_multiplier_stages([2, 3, 4])

	input_data = [10, 20]
	output = pipeline.process(input_data)

	assert output[0] == [20, 40], f"Kļūda: Nepareizs reizinātājs! Iegūts {output[0]}"
	print("Scenārijs 2 darbojas veiksmīgi.")